### PR TITLE
Switch to copier template

### PR DESCRIPTION
**Pull request by nvaytet**

### CODE REVIEW

File: conda/meta.yaml [hunk 1]
```diff
@@ -12,10 +14,8 @@ requirements:
     - setuptools
     - setuptools_scm
   run:
-    - python >=3.8
-    - python-dateutil
     - matplotlib
-  run_constrained:
+    - python>=3.8
```
Comment (jl-wynen) [inline conda/meta.yaml:18]: The min version is 3.9 in other places.

File: pyproject.toml [hunk 2]
```diff
@@ -1,46 +1,84 @@
...
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Topic :: Scientific/Engineering",
+    "Typing :: Typed",
+]
+requires-python = ">=3.8"
```
Comment (jl-wynen) [inline pyproject.toml:28]: Also 3.8

File: .github/workflows/python-version-ci [hunk 3]
```diff
@@ -0,0 +1 @@
+3.9
```
Comment (jl-wynen) [inline .github/workflows/python-version-ci:1]: Here it's 3.9

File: requirements/all.in
Comment (jl-wynen) [inline requirements/all.in:1]: What is the purpose of this file? And why did you add it here but not in the template?

File: tox.ini [hunk 4]
```diff
@@ -1,44 +1,79 @@
 [tox]
-envlist = minimal,noplotly,py310
+envlist = py39
```
Comment (jl-wynen) [inline tox.ini:2]: And here it's 3.9

File: pyproject.toml [hunk 5]
```diff
@@ -1,46 +1,84 @@
...
+license = { file = "LICENSE" }
+readme = "README.md"
+classifiers = [
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: BSD License",
+    "Natural Language :: English",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
```
Comment (jl-wynen) [inline pyproject.toml:21]: 3.8

File: conda/meta.yaml (hunk 1 above)
Comment (nvaytet) [reply conda/meta.yaml:18]: 3.8/3.9 is because I asked for 3.8 when answering the questions of copier, but then was having issues on CI with the env generated by `pip-compile-multi`.

I think it's because I ran `pip-compile-multi` on my machine that has 3.10, and there is some conflict with a library `contourpy` used by `matplotlib`: see [here](https://github.com/scipp/plopp/actions/runs/6961743750/job/18944021427).

So I was lazy, didn't want to start the whole copier thing from scratch, so I tried to change it manually. I obviously forgot quite a few places, meaning that I probably should not have been lazy...

Anyway, thanks for being thorough.

File: requirements/all.in
Comment (nvaytet) [reply requirements/all.in:1]: The only way I managed to get the `pip-compile-multi` to work is by adding those pins everywhere.
I'm hoping there is a better way?

It felt hacky and experimental, so I didn't want to add it to the template just yet.

File: conda/meta.yaml (hunk 1 above)
Comment (jl-wynen) [reply conda/meta.yaml:18]: It's fine to switch to 3.9 now, I think. Running copier again should be quick, you only need to change one answer.

File: requirements/all.in
Comment (nvaytet) [reply requirements/all.in:1]: I think I am having the same issue here: https://github.com/scipp/copier_template/pull/103

**Comment by nvaytet** [comment]
@jl-wynen I think this is ready for another quick look. Thanks!

File: requirements/all.in
Comment (jl-wynen) [reply requirements/all.in:1]: Yes, I also keep running into this locally.

Maybe the `--autoresolve` option of `pip-compile-multi` can help. I haven't tried it...

File: docs/index.rst
Comment (jl-wynen) [inline docs/index.rst:1]: Do you want to stick with rst here instead of md? This may make it harder down the line as the template gets updated.

File: requirements/all.in
Comment (nvaytet) [reply requirements/all.in:1]: I will try :+1:

Comment (nvaytet) [reply requirements/all.in:1]: See https://github.com/scipp/copier_template/pull/106

**Review by jl-wynen — APPROVED** [review]
Looks good. You can either merge now or wait for the change in the template.
